File: TP/JPA-HIBERNATE/eshop-jpa-correction/src/main/java/fr/formation/model/Fournisseur.java
package fr.formation.model;

import java.util.List;

import jakarta.persistence.AttributeOverride;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;

@Entity
@Table(name = "fournisseur")
@AttributeOverride(name = "id", column = @Column(name = "fou_id"))
@AttributeOverride(name = "nom", column = @Column(name = "fou_nom", length = 100, nullable = false))
public class Fournisseur extends Personne {
	@Column(name = "fou_responsable", length = 200, nullable = false)
	private String responsable;

	@OneToMany(mappedBy = "fournisseur")
	private List<Produit> produits;

	public String getResponsable() {
		return responsable;
	}

	public void setResponsable(String responsable) {
		this.responsable = responsable;
	}

	public List<Produit> getProduits() {
		return produits;
	}

	public void setProduits(List<Produit> produits) {
		this.produits = produits;
	}
}

File: TP/ANGULAR/eshop-angular/src/app/fournisseur.ts
export class Fournisseur {
  public get id(): number {
    return this._id;
  }

  public set id(value: number) {
    this._id = value;
  }
  
  public get nom(): string {
    return this._nom;
  }

  public set nom(value: string) {
    this._nom = value;
  }

  public get responsable(): string {
    return this._responsable;
  }
  
  public set responsable(value: string) {
    this._responsable = value;
  }
  
  constructor(
    private _id: number,
    private _nom: string,
    private _responsable: string) { }
}
